ui/ui-frontend/projects/vitamui-library/src/app/modules/components/file-selector/file-selector.component.spec.ts

```diff
@@ -64,4 +65,63 @@ describe('FileSelectorComponent', () => {
     component.handleFilesSelection(mockFiles);
     expect(component['inputFiles'].nativeElement.value).toBe('');
   });
+
+  it('should filter files based on allowed extensions', () => {
+    const file1 = new File([''], 'file1.json', { type: 'application/json' });
+    const file2 = new File([''], 'file2.txt', { type: 'text/plain' });
+    component.extensions = ['.json'];
+    component.handleFilesSelection([file1, file2] as any);
+    expect(component.files.length).toBe(1);
+    expect(component.files[0].name).toBe('file1.json');
+  });
+
+  it('should limit files to one if multipleFiles is false', () => {
+    const file1 = new File([''], 'file1.json', { type: 'application/json' });
+    const file2 = new File([''], 'file2.json', { type: 'application/json' });
+    component.multipleFiles = false;
+    component.handleFilesSelection([file1, file2] as any);
```

Comment:
` as any` inutile, à retirer partout.

```suggestion
    component.handleFilesSelection([file1, file2]);
```